src/components/Calculator.js

```diff
@@ -1,35 +1,50 @@
+// Calculator.js
 import React from 'react';
+import calculate from '../logic/calculate';
+import Button from './Button'; // Assuming Button.js is in the same directory as Calculator.js
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
     this.state = {};
   }
 
+  clickHandler = (e) => {
+    const result = calculate(this.state, e.target.textContent);
+    this.setState(result);
+  }
```

Comment:
- [ ] [Optional] I would suggest you please use the functional component in place of the class-based component.